lab_5_scrapper/scrapper.py

```diff
@@ -120,38 +242,101 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Finds text of article
         """
-        pass
+        article_text = ""
+        paragraphs = article_soup.find_all("div", class_="article__paragraph")
+        for p in paragraphs:
+            texts = p.find_all("p")
+            for text in texts:
+                article_text += text.text.strip() + " "
+        self.text = article_text.strip()
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        # id
+        id_tag = article_soup.find('meta', {'property': 'og:url'})
+        if id_tag:
+            self.article.id = id_tag['content'].split('/')[-1]
+        else:
+            self.article.id = 'NOT FOUND'
+
+        # title
+        title_tag = article_soup.find('meta', {'property': 'og:title'})
+        if title_tag:
+            self.article.title = title_tag['content']
+        else:
+            self.article.title = 'NOT FOUND'
+
+        # author
+        author_tag = article_soup.find('div', {'class': 'article__footer'}).find('p', {'class': 'article__prepared'})
+        if author_tag:
+            self.article.authors = [author_tag.text.strip()]
+        else:
+            self.article.authors = ['NOT FOUND']
+
+        # publication date
+        date_str = article_soup.find("div", class_="time news-footer__time").string.strip()
+        self.article.publication_date = self.unify_date_format(date_str)
+
+
+
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        month_dict = {'янв': '01', 'фев': '02', 'мар': '03', 'апр': '04', 'май': '05', 'июн': '06',
+                      'июл': '07', 'авг': '08', 'сен': '09', 'окт': '10', 'ноя': '11', 'дек': '12'}
+
+        date_str = date_str.strip()
+
+        for month, value in month_dict.items():
+            date_str = date_str.replace(month, value)
+
+        date_format = '%Y-%m-%d %H:%M:%S'
+        try:
+            date = datetime.datetime.strptime(date_str, date_format)
+        except ValueError:
+            date = None
+
+        return date
+
+
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = requests.get(self.article_url)
+        response.raise_for_status()
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+
+        return self.article
+
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    base_path = Path(base_path)
+
+    if base_path.exists() and any(base_path.iterdir()):
+        shutil.rmtree(base_path)
+
+    base_path.mkdir(parents=True, exist_ok=True)
 
```

Comment:
do not forget to show how you run all these things and also demonstrate how to collect 100 articles for your website